gcloud-java-storage/src/main/java/com/google/cloud/storage/testing/FakeStorageRpc.java

```diff
@@ -37,9 +37,24 @@
 
 /**
  * A bare-bones in-memory implementation of Storage, meant for testing.
- * See LocalGcsHelper.
  *
- * This class is NOT thread-safe.
+ * <p>This class is <i>not</i> thread-safe. It's also (currently) limited in the following ways:
+ * <ul>
+ * <li>Partially Supported
+ *   <ul>
+ *   <li>create
+ *   <li>get
+ *   <li>delete
+ *   <li>directory listing
+ *   </ul>
+ * <li>Unsupported
```

Comment:
It seems to me that also bucket operations (get, list, delete), batches and compose are not supported

Not related to this PR: I might be missing something but why do some methods throw UnsupportedOperationException (e.g. createBatch) while others return null (e.g. compose)?